account_tax_balance/views/account_move_view.xml

```diff
@@ -18,7 +18,10 @@
         <field name="inherit_id" ref="account.view_move_form" />
         <field name="arch" type="xml">
             <field name="ref" position="after">
-                <field name="financial_type" />
+                 <field
+                    name="financial_type"
+                    invisible="context.get('default_move_type') in ['out_invoice','in_invoice','out_refund','in_refund']"
```

Comment:
I think it's too complicated using context on this. `move_type` field is always present on the `account.view_move_form`. Maybe simplify like this:
```suggestion
                    attrs="{'invisible': [('move_type', 'in', ['out_invoice','in_invoice','out_refund','in_refund'])]}"
```